src/main/java/org/victorrobotics/frc/dtlib/hardware/phoenix5/DTTalonSRX.java

```diff
@@ -0,0 +1,187 @@
+  }
+
+  @Override
+  public void configOutputInverted(boolean invert) {
+    internal.setInverted(invert);
+  }
+
+  @Override
+  public void configOpenLoopRampRate(double seconds0To100) {
+    internal.configOpenloopRamp(seconds0To100);
+  }
+
+  @Override
+  public void configClosedLoopRampRate(double seconds0To100) {
+    internal.configClosedloopRamp(seconds0To100);
+  }
+
+  @Override
+  public void configPID(int slot, double proportional, double integral, double derivative, double velocityFF,
+      double staticFF, double integralZone) {
+    if (slot < 0 || slot > 3) {
+      throw new DTIllegalArgumentException(slot, "slot must be in range 0-3");
+    }
+
+    // No velocityFF
+    if (!Double.isNaN(proportional)) {
+      internal.config_kP(slot, proportional);
+    }
+    if (!Double.isNaN(integral)) {
+      internal.config_kI(slot, integral);
+    }
+    if (!Double.isNaN(derivative)) {
+      internal.config_kD(slot, derivative);
+    }
+    if (!Double.isNaN(staticFF)) {
+      internal.config_kF(slot, staticFF);
+    }
+    if (!Double.isNaN(integralZone)) {
+      internal.config_IntegralZone(slot, integralZone);
+    }
+  }
+
+  @Override
+  public void setPIDSlot(int slot) {
+    internal.selectProfileSlot(slot, 0);
+  }
+
+  @Override
+  public double[] getPIDConstants(int slot) {
+    // Can't read constants from TalonSRX; cache?
+    return new double[6];
+  }
+
+  @Override
+  public void configCurrentLimit(int maxSupplyCurrent) {
+    internal.configSupplyCurrentLimit(new SupplyCurrentLimitConfiguration(true, maxSupplyCurrent, maxSupplyCurrent, 0));
+  }
+
+  @Override
+  public boolean isOutputInverted() {
+    return internal.getInverted();
+  }
+
+  @Override
+  public void setPercentOutput(double percent) {
+    internal.set(TalonSRXControlMode.PercentOutput, percent);
+  }
+
+  @Override
+  public void setPosition(double position) {
+    // TODO Auto-generated method stub
+    throw new UnsupportedOperationException("Unimplemented method 'setPosition'");
+  }
+
+  @Override
+  public void setVelocity(double velocity) {
+    // TODO Auto-generated method stub
+    throw new UnsupportedOperationException("Unimplemented method 'setVelocity'");
+  }
+
+  @Override
+  public void neutralOutput() {
+    internal.neutralOutput();
+  }
+
+  @Override
+  public void setEncoderPosition(double position) {
+    // TODO Auto-generated method stub
+    throw new UnsupportedOperationException("Unimplemented method 'setEncoderPosition'");
+  }
+
+  @Override
+  public double getMotorOutputPercent() {
+    return internal.getMotorOutputPercent();
+  }
+
+  @Override
+  public double getInputVoltage() {
+    return internal.getBusVoltage();
+  }
+
+  @Override
+  public double getTemperature() {
+    return internal.getTemperature();
+  }
+
+  @Override
+  public double getEncoderPosition() {
+    // TODO Auto-generated method stub
+    throw new UnsupportedOperationException("Unimplemented method 'getEncoderPosition'");
```

Comment:
I'm generally not a big fan of throwing for things like this, but in this case I'll make an exception

Reply:
(the alternative would be to have a base motor speed controller interface, and extend that for a motor controller with position/velocity feedback)

Reply:
That is worth looking into. I don't like unsupported operations either, but for the moment it is, in fact, an unsupported operation.